pandas/core/base.py

```diff
@@ -512,10 +517,24 @@ def _make_str_accessor(self):
         elif isinstance(self, Index) and self.inferred_type != 'string':
             raise AttributeError("Can only use .str accessor with string "
                                  "values (i.e. inferred_type is 'string')")
+
+    # string methods
+    def _make_str_accessor(self):
+        self._check_str_accessor()
         return StringMethods(self)
 
     str = AccessorProperty(StringMethods, _make_str_accessor)
 
+    def _dir_additions(self):
+        return set()
+
+    def _dir_deletions(self):
+        try:
+            self._check_str_accessor()
```

Comment:
I did that to avoid having to instantiate `StringMethods(self)` when it is not needed, but I agree it's probably premature optimization